app/controllers/courses_controller.rb

```diff
@@ -56,11 +56,24 @@ def destroy
     end
 
     def flashcard
+        # get the user and course paramaters
         @user = User.find(params[:user_id])
         proper_user(@user)
         @course = @user.courses.find(params[:id])
+        # randomly select one student from the course roster
         @student = @course.students.sample
-        render(json:  @student.to_json)
+        # if there is a student to show
+        if @student != nil
+            # convert the student object to a hash
+            @student_hash = @student.attributes
+            # add the student's portrait url to the hash
+            @student_hash[:portrait_url] = @student.portrait.url(:flashcard)
```

Comment:
Out of curiosity, why do you need this line? Is the portrait_url not included in the student attributes?

Reply:
For some reason no. It's an attachment, so I suspect it comes in from another database table.

Reply:
Hmmm... could this cause problems down the line, or is it just a quirk?

Reply:
I think it's better this way, we don't want image changes to affect the student table. Also, I think it's paperclip, so we might not be able to change it.